server/src/main/java/org/elasticsearch/health/node/ShardsCapacityHealthIndicatorService.java

```diff
@@ -89,12 +97,96 @@ public class ShardsCapacityHealthIndicatorService implements HealthIndicatorServ
         "frozen"
     );
 
+    public static final NodeFeature SHARD_CAPACITY_UNHEALTHY_THRESHOLD_SETTINGS = new NodeFeature(
```

Comment:
Nit: can we include `_FEATURE` somewhere in the field name to avoid confusion between this field and the actual settings?